open-sphere-base/core/src/main/java/io/opensphere/core/pipeline/processor/TextureProcessor.java

```diff
@@ -917,19 +918,34 @@ protected void processImageLoaded(Collection<? extends E> objects, StateControll
         Collection<E> textureDataLoaded = New.collection(objects.size());
         Collection<E> textureLoaded = New.collection(0);
         Collection<E> reload = null;
+        Collection<E> retFailed = null;
+
         for (E geom : objects)
         {
             handleBlankImage(geom);
             ImageGroup imageData = geom.getImageManager().pollCachedImageData();
+
             if (imageData == null)
             {
                 reload = handleMissingImageData(geom, textureLoaded, reload);
             }
             else
             {
                 // Ignore failed here because there may be another thread trying
-                // to process the image.
-                handleImageLoaded(textureDataLoaded, textureLoaded, (Collection<E>)null, geom, imageData);
+                // to process the image. If failed is instance of
+                // PointSpriteGeometry, reprocess.
+
+                retFailed = handleImageLoaded(textureDataLoaded, textureLoaded, (Collection<E>)null, geom, imageData);
+                if (retFailed != null)
+                {
+                    retFailed.forEach(geometry ->
+                    {
+                        if (geometry instanceof PointSpriteGeometry)
+                        {
+                            handleImageLoaded(textureDataLoaded, textureLoaded, (Collection<E>)null, geometry, imageData);
+                        }
+                    });
+                }

```

Comment:
This is up to you, but you can collapse this a bit further, down to one line:

retFailed.stream().filter(geometry -> geometry instanceof PointSpriteGeometry).forEach(geometry -> handleImageLoaded(textureDataLoaded, textureLoaded, (Collection<E>)null, geometry, imageData));